tutorials/W2D5_ClimateResponse-AdaptationImpact/W2D5_Tutorial5.ipynb

```diff
@@ -0,0 +1,1041 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 607
+    },
+    "id": "osyOejNO5CZE",
+    "outputId": "9d70f7cf-cd11-4d5a-d8c6-d2903e01798c"
+   },
+   "outputs": [],
+   "source": [
+    "# @title  { display-mode: \"form\" }\n",
+    "# @markdown Run this cell to see the plot of results!\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "# Extract model names and average scores from results\n",
+    "model_names = list(results.keys())\n",
+    "average_scores = [results[model_name][\"Average Score\"] for model_name in model_names]\n",
+    "\n",
+    "# Adjust scores to plot zero for negative scores\n",
+    "adjusted_scores = [score if score > 0 else 0 for score in average_scores]\n",
+    "\n",
+    "# Plotting\n",
+    "plt.figure(figsize=(10, 6))\n",
+    "plt.bar(model_names, adjusted_scores, color=['skyblue' if score > 0 else 'lightgray' for score in average_scores])\n",
+    "plt.xlabel('Model')\n",
+    "plt.ylabel('Average Score')\n",
+    "plt.title('Average Score of Different Regression Models')\n",
+    "plt.xticks(rotation=45, ha='right')  # Rotate x-axis labels for better readability\n",
+    "plt.tight_layout()\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "E8MVfXNSAFWc"
+   },
+   "source": [
+    "This quick sweep of models suggests Random Forest is a good choice, but recall that most of these models have hyperparameters. Varying these hyperparameters may lead to different results!\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "C0Ujuy7zDcfg"
+   },
+   "source": [
+    "\n",
+    "\n",
+    "\n",
+    "---\n",
+    "\n",
+    "\n",
+    "                                Congratulations! You have reached the end of the  tutorial.   \n",
+    "\n",
+    "\n",
+    "\n",
+    "---\n"
+   ]
+  }
+ ],
+ "metadata": {
+  "colab": {
+   "provenance": []
+  },
+  "kernelspec": {
+   "display_name": "Python 3 (ipykernel)",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.9.0"
+  },
+  "widgets": {
+   "application/vnd.jupyter.widget-state+json": {
+    "05aafe4131a14326ad2d835644586e5f": {
+     "model_module": "@jupyter-widgets/output",
+     "model_module_version": "1.0.0",
+     "model_name": "OutputModel",
+     "state": {
+      "_dom_classes": [],
+      "_model_module": "@jupyter-widgets/output",
+      "_model_module_version": "1.0.0",
+      "_model_name": "OutputModel",
+      "_view_count": null,
+      "_view_module": "@jupyter-widgets/output",
+      "_view_module_version": "1.0.0",
```

Comment:
same question

Reply:
same , replaced with the code as given in template